src/OrasProject.Oras/Registry/Remote/ManifestStore.cs

```diff
@@ -133,17 +133,20 @@ public async Task<bool> ExistsAsync(Descriptor target, CancellationToken cancell
     }
 
     /// <summary>
-    /// Pushes the content, matching the expected descriptor.
+    /// Pushes the manifest content, matching the expected descriptor.
     /// </summary>
     /// <param name="expected"></param>
     /// <param name="content"></param>
     /// <param name="cancellationToken"></param>
     /// <returns></returns>
     public async Task PushAsync(Descriptor expected, Stream content, CancellationToken cancellationToken = default)
-        => await InternalPushAsync(expected, content, expected.Digest, cancellationToken).ConfigureAwait(false);
+    {
+        var remoteReference = Repository.ParseReference(expected.Digest);
+        await DoPushAsync(expected, content, remoteReference, cancellationToken).ConfigureAwait(false);
+    }
```

Comment:
Will it simply work?
```suggestion
        => await PushAsync(expected, content, expected.Digest, cancellationToken).ConfigureAwait(false);
```

Reply:
Yeah I missed that, updated.